common/djangoapps/track/views/segmentio.py

```diff
@@ -155,11 +158,22 @@ def track_segmentio_event(request):  # pylint: disable=too-many-statements
     if any(disallowed_subs_name in segment_event_name.lower() for disallowed_subs_name in disallowed_substring_names):
         return
 
+    # Attempt to extract and validate the data field.
     if 'data' not in segment_properties:
         raise EventValidationError(ERROR_MISSING_DATA)
+    segment_event_data = segment_properties.get('data', {})
+    if type(segment_event_data) is not dict:
+        set_custom_attribute('segment_unexpected_data', str(segment_event_data))
+        raise EventValidationError(ERROR_INVALID_DATA_FIELD_TYPE)
+
+    set_custom_attribute('segment_event_name', segment_event_name)
+    set_custom_attribute('segment_event_source', event_source)
```

Comment:
You should move these 2 lines before you start raising errors so you'll have them as additional data on the errors and transactions in New Relic.

Also, not sure if `segment_event_type` would also be useful in a custom attribute for debugging purposes.

Reply:
segment_event_type seems to always be "track" (see https://github.com/edx/edx-platform/blob/b8bd4f63/lms/envs/common.py#L1154) so I assume it isn't useful to capture.